pkg/services/ngalert/schedule/schedule.go

```diff
@@ -380,15 +380,16 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 		ruleEval, err := sch.evaluatorFactory.Create(evalCtx, e.rule.GetEvalCondition())
 		var results eval.Results
 		var dur time.Duration
-		if err == nil {
+		if err != nil {
+			dur = sch.clock.Now().Sub(start)
```

Comment:
How come we assign `dur` to the same value in both branches?

Reply:
Discussed this offline - we need to have due in both branches because they depend on each other (otherwise we won't capture the evaluation time).